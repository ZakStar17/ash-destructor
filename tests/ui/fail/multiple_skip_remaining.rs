use ash_destructor::DeviceDestroyable;

#[path = "../../utils/mod.rs"]
mod utils;

use utils::ImplDeviceDestroyable;

#[derive(DeviceDestroyable)]
struct Named {
    #[destroy_ignore]
    a: ImplDeviceDestroyable,
    #[destroy_ignore_remaining]
    b: String,
    c: u32,
    #[destroy_ignore_remaining]
    d: bool,
}

#[derive(DeviceDestroyable)]
struct Unnamed(
    ImplDeviceDestroyable,
    #[destroy_ignore_remaining]
    #[destroy_ignore_remaining]
    String,
);

fn main() {}
