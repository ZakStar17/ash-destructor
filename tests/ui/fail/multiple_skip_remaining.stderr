error: Multiple #[destroy_ignore_remaining] attributes in "Named"
  --> tests/ui/fail/multiple_skip_remaining.rs:15:5
   |
15 |     #[destroy_ignore_remaining]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Multiple #[destroy_ignore_remaining] attributes in "Unnamed"
  --> tests/ui/fail/multiple_skip_remaining.rs:23:5
   |
23 |     #[destroy_ignore_remaining]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
